app/models/compute_resource.rb

```diff
@@ -415,6 +415,10 @@ def nested_attributes_for(type, opts)
   end
 
   def associate_by(name, attributes)
+    if name == "mac"
+      Net::Validations.normalize_mac(attributes) if attributes.kind_of? String
+      attributes = attributes.map { |mac| Net::Validations.normalize_mac(mac) } unless attributes.kind_of? String
```

Comment:
you could use this instead of whole `if` block to make this change a one liner.
```suggestion
      attributes = Array.wrap(attributes).map { |mac| Net::Validations.normalize_mac(mac) } if name == 'mac'
```
It would become always array, but `ActiveRecord` can handle sindle valued arrays  as a single value internally.

Reply:
Thanks for hint @ezr-ondrej